services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/events.py

```diff
@@ -551,59 +557,52 @@ async def _remove_containers_save_state_and_outputs() -> None:
                 logger.warning(
                     (
-                        "Could not contact dynamic-sidecar to begin destruction of "
+                        "There was an issue while removing contains for "
                         "%s\n%s. Will continue service removal!"
                     ),
                     scheduler_data.service_name,
                     f"{e}",
                 )
 
             # only try to save the status if :
-            # - the dynamic-sidecar has finished booting correctly
             # - it is requested to save the state
+            # - the dynamic-sidecar has finished booting correctly
             if (
                 scheduler_data.dynamic_sidecar.service_removal_state.can_save
-                and await are_all_services_present(
-                    node_uuid=scheduler_data.node_uuid,
-                    dynamic_sidecar_settings=app.state.settings.DYNAMIC_SERVICES.DYNAMIC_SIDECAR,
-                )
+                and scheduler_data.dynamic_sidecar.were_services_created
             ):
                 dynamic_sidecar_client = get_dynamic_sidecar_client(app)
-                dynamic_sidecar_endpoint = scheduler_data.dynamic_sidecar.endpoint
 
                 logger.info(
                     "Calling into dynamic-sidecar to save state and pushing data to nodeports"
                 )
                 try:
                     tasks = [
-                        dynamic_sidecar_client.service_push_output_ports(
-                            dynamic_sidecar_endpoint,
+                        dynamic_sidecar_client.ports_outputs_push(
+                            dynamic_sidecar_endpoint
                         )
                     ]
+
                     # When enabled no longer uploads state via nodeports
                     # It uses rclone mounted volumes for this task.
                     if not app_settings.DIRECTOR_V2_DEV_FEATURES_ENABLED:
                         tasks.append(
-                            dynamic_sidecar_client.service_save_state(
-                                dynamic_sidecar_endpoint,
-                            )
+                            dynamic_sidecar_client.state_save(dynamic_sidecar_endpoint)
                         )
 
-                    try:
-                        # at most 2 parallel tasks will be running
-                        await logged_gather(*tasks, max_concurrency=2)
-                    except NodeportsDidNotFindNodeError as err:
-                        logger.warning("%s", f"{err}")
+                    await logged_gather(*tasks, max_concurrency=2)
 
                     logger.info("Ports data pushed by dynamic-sidecar")
-                except BaseClientHTTPError as e:
+                except (BaseClientHTTPError, TaskClientResultError) as e:
                     logger.warning(
                         (
                             "Could not contact dynamic-sidecar to save service "
```

Comment:
here, this should be an error no? I mean it is unexpected. Also it may be either the state saving or the outputs pushing that failed or both. Maybe you should check the returned error and differentiate the error so whoever needs to manually intervene has a better idea.

Reply:
This has little consequence. Usually it happens because the containers are gone and you want to remove them.
I would still leave it as a warning and maybe have some better text around it. I've tried to word it better.